projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func Execute() {
+	// define a flag of type boolean with the specific properties (name, value, usage)
+	helpFlag := flag.Bool("h", false, "show help")
+
+	// parse the command line flags
+	flag.Parse()
+
+	// if the "-h" flag is provided
+	if *helpFlag {
+		fmt.Fprintf(os.Stdout, "go-ls help message\n")
+		return
+	}
+
+	// initialise/default the directory path to where go-ls was called from
+	directoryPath, err := os.Getwd()
+
+	// if we can't get the current working directory
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get the current working directory: %v\n", err)
+		os.Exit(2)
+	}
+
+	// if go-ls was called with an argument use that as the directory path instead
+	if len(os.Args) > 1 {
```

Comment:
What happens to the other args if I passed more than 1 arg? Right now it looks like you're just ignoring them? In general it's not very friendly to just silently ignore user input - either we should handle it, or we should tell the user they've supplied information we don't expect.

This is particularly important if we're expecting our program to be run in scripts or other automated locations, and particularly if we _change_ things over time. Imagine our program had a `--fail-on-invalid-input` flag, the user was setting it, then our program started ignoring it - this would be bad - a user would have thought they'd opted in to some kind of validation, and isn't actually getting it.

Reply:
I have tried to rewrite it so it can handle multiple arguments and will behave similarly to the real `ls`